src/briefcase/integrations/docker.py

```diff
@@ -378,9 +381,377 @@ def check_output(self, args: list[SubprocessArgT], image_tag: str) -> str:
+        if self._x11_is_display_tcp(display_num):
+            proxy_process = self.tools.subprocess.Popen(
+                [
+                    "socat",
+                    f"TCP-LISTEN:{proxy_display_tcp_port},reuseaddr,fork,bind=0.0.0.0",
+                    f"TCP:localhost:{6000 + display_num}",
+                ]
+            )
+
+        # Otherwise, just proxy to the UNIX socket for the display
+        elif self._x11_is_display_unix(display_num):
+            proxy_process = self.tools.subprocess.Popen(
+                [
+                    "socat",
+                    f"TCP-LISTEN:{proxy_display_tcp_port},reuseaddr,fork,bind=0.0.0.0",
+                    f"UNIX-CONNECT:/tmp/.X11-unix/X{display_num}",
+                ]
+            )
+
+        else:
+            raise BriefcaseCommandError(f"Cannot find X11 display for {DISPLAY!r}")
+
+        return proxy_process, proxy_display_num
+
+    def _x11_proxy_display_xauth_file_path(
+        self, display_num: int
+    ) -> PosixPath:  # pragma: no-cover-if-is-windows
+        """Path to the xauth database for the proxy display.
+
+        The project's build directory is used to accommodate Docker Desktop. By default,
+        Docker Desktop only allows bind mounts from directories/files within the user's
+        home directory. So, while it may be more appropriate to use a dedicated temp
+        directory, such as /tmp, this avoids having to instruct users to add that
+        directory as a source of possible bind mounts.
+        """
+        return PosixPath.cwd() / f"build/.briefcase.docker.xauth.{display_num}"
+
+    def _x11_allocate_display(self) -> int:
+        """Finds available X display number for use; raises otherwise.
+
+        A universal mechanism to request and reserve an X display does not exist. On
+        most systems, though, there will not be many active X displays. So, this
+        approach is similar to that of OpenSSH's strategy for reserving an X display
+        for X11 forwarding: simply iterate over candidate display numbers using the
+        first one that is not already is use.
+
+        This check for whether a display number is already in use is limited to
+        verifying a TCP server or UNIX socket does not already exist for the display.
+        While other transport protocols are possible, any modern system should be using
+        at least one of these transports in tandem with anything else.
+        """
+        for num in range(20, 167 + 1, 3):
+            if not (self._x11_is_display_unix(num) or self._x11_is_display_tcp(num)):
+                display_num = num
+                break
+        else:
+            raise BriefcaseCommandError("Failed to allocate X11 display")
+
+        return display_num
+
+    def _x11_is_display_tcp(self, display_num: int) -> bool:
+        """Is a TCP server running for the X display?"""
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
+            sock.settimeout(3)
```

Comment:
If I'm reading this right, the pathological case could take ~150s to resolve that there's no available display. I presume this isn't likely in practice... what *is* a likely delay? Would it be worth putting the display discovery in a waitbar?

Reply:
Interestingly...when I first implemented this, I forgot to include a `break`; so, for days, it was always connecting to all 50 ports and returning the last one when the loop ended. The loop would run so fast, I never even noticed.

So, setting a timeout at all is just to ensure this doesn't block indefinitely on a port that's completely misbehaving.

The truly pathological case where all 50 ports block for 3 seconds....that'd be interesting. I think that would mean "something" is listening....but not responding. Because if nothing is listening on that port, the underlying OS should immediately reject the connection attempt (I'd expect anyway).

Sooo, I'm not completely against adding a Wait Bar....but setting up and tearing down the Wait Bar would probably take longer than the 99.9% case where the first connection works and is returned.

Reply:
Ok - if the "full timeout" case is borderline impossible, I'm OK without a waitbar here.